aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,312 @@
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NamespaceNotEmptyException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.CreateDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.DeleteDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.DeleteTableRequest;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabaseResponse;
+import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
+import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
+import software.amazon.awssdk.services.glue.model.GetTablesRequest;
+import software.amazon.awssdk.services.glue.model.GetTablesResponse;
+import software.amazon.awssdk.services.glue.model.InvalidInputException;
+import software.amazon.awssdk.services.glue.model.UpdateDatabaseRequest;
+
+public class GlueCatalog extends BaseMetastoreCatalog implements Closeable, SupportsNamespaces {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueCatalog.class);
+
+  private final Configuration conf;
+  private final GlueClient glue;
+  // Catalog ID is the AWS account ID. When it is null, it uses the account of the Glue client.
+  private final String catalogId;
+  private final String warehousePath;
+
+  /**
+   * constructor with default glue client
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, String catalogId, String warehousePath) {
+    this(conf, GlueClient.create(), catalogId, warehousePath);
+  }
+
+  /**
+   * constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   */
+  public GlueCatalog(Configuration conf, GlueClient glue, String catalogId, String warehousePath) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.warehousePath = warehousePath;
```

Comment:
Nit: Can we add `Preconditions.checkNotNull` on anything that is nullable here so we can avoid nulll checks later?

Reply:
warehousePath is checked by `validateWarehousePath` in the constructor, and `catalogId` can be null. I can add the precondition for `conf` and `glue`.